### DISCUSSION TITLE
JsonPatch with System.Text.Json

### DISCUSSION THREAD

**Discussion by Shirasho** [dotnet/runtime#43226] [Ideas]
Have there been any discussions implementing JsonPatch with System.Text.Json? The feature is implemented with Json.NET in the aspnetcore repo (https://github.com/dotnet/aspnetcore/tree/master/src/Features/JsonPatch/src), but with the release of .NET Core 3.0 we have migrated to System.Text.Json.

Our platform has a requirement to detect changes in a large payload, and right now we do that by hand using custom logic and a custom format. Our backend is now implementing JsonPatch for this functionality, but our client has no way to use this functionality without adding the Json.NET dependency back into our application.

We have the option of trying to port the aspnetcore implementation to use System.Text.Json ourselves, but we do not want the burden of maintaining this.

**Comment by eerhardt** [comment]
cc @steveharter @layomia

**Comment by layomia** [comment]
> Our backend is now implementing JsonPatch for this functionality, but our client has no way to use this functionality without adding the Json.NET dependency back into our application.

@shirasho, is it possible to share your System.Text.Json-based implementation?

The best way to move this proposal forward is to put together an API proposal using this link https://github.com/dotnet/runtime/issues/new?labels=api-suggestion,area-System.Text.Json&template=02_api_proposal.md. The goal here is so that we can fully understand the feature and how it will be used, and then ultimately it can be reviewed.

An API proposal would also be good a good way to get a ref count from the community so we get a sense of how impactful this feature would be.

**Comment by AlbertoPa** [comment]
I would say reproducing the functionality described here 

https://docs.microsoft.com/en-us/aspnet/core/web-api/jsonpatch?view=aspnetcore-6.0#:~:text=JSON%20Patch%20is%20a%20format%20for%20specifying%20updates,an%20array%20element%20or%20replacing%20a%20property%20value.

would be a good start.

Currently the situation with System.Text.Json is that it does not support certain operations commonly encountered in APIs. Additionally, in some parts of .Net (e.g., Blazor), the HTTP client extension methods do not easily allow to drop in Newtonsoft as a replacement without extra custom code (in addition to the extra payload for the client app), which may be an issue if a need to use PATCH operations presents.
